### PR TITLE
ParseEvaluateResult should always match with a single result

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Review by kblok — APPROVED** [review]

File: src/PlaywrightSharp/ScriptsHelper.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ internal static object ParseEvaluateResult(JsonElement? element, Type t)
         {
             var parseEvaluateResult = typeof(ScriptsHelper)
                 .GetMethods(BindingFlags.NonPublic | BindingFlags.Static)
-                .FirstOrDefault(m => m.Name == "ParseEvaluateResult" && m.GetGenericArguments().Any());
+                .Single(m => m.Name == "ParseEvaluateResult" && m.GetGenericArguments().Any());
```
Comment (SimonCropp) [inline src/PlaywrightSharp/ScriptsHelper.cs:47]: also: isnt isgeneric better than `GetGenericArguments().Any(`?

Comment (kblok) [reply src/PlaywrightSharp/ScriptsHelper.cs:47]: Could you try implementing that?

Comment (SimonCropp) [reply src/PlaywrightSharp/ScriptsHelper.cs]: done

Comment (SimonCropp) [reply src/PlaywrightSharp/ScriptsHelper.cs]: I cached the reflected value statically

**Pull request merged**